lib/environment/foreground/ajax.js

```diff
@@ -112,11 +112,16 @@ function buildRequestParams({ method = 'GET', url, query = {}, headers = {}, dat
 		urlObj.searchParams.set(key, String(val));
 	}
 
-	const sameOrigin = urlObj.hostname.includes(location.hostname.split('.').slice(-2).join('.'));
-	if (sameOrigin) {
+	const sameSite = urlObj.hostname.includes(location.hostname.split('.').slice(-2).join('.'));
```

Comment:
With that `.slice(-2)` here, this may as well read `urlObj.hostname.endsWith('.reddit.com')`. Maybe pull the `'.reddit.com'` magic value from the manifest.